### PR TITLE
[IHOST-1625] Remote entities

### PR DESCRIPTION
**Pull request by cristianciutea**
#### Description of the changes

This PR will add remote entities monitoring

https://newrelic.atlassian.net/browse/IHOST-1625

### Author

- [ ] add a risk label after carefully considering the "blast radius" of your changes
- [ ] describe the _intent_ of your changes in the description. don't just rewrite your code in prose
- [ ] assign at least one reviewer

### Reviewer

- [ ] review code for readability
- [ ] verify that high risk behavior changes are well tested
- [ ] check license for any new external dependency
- [ ] ask questions about anything that isn't clear and obvious
- [ ] approve the PR when you consider it's good to merge

### CODE REVIEW

File: src/apache.go [hunk 1]
```diff
@@ -11,16 +17,24 @@ import (
...
 }
 
 const (
 	integrationName    = "com.newrelic.apache"
-	integrationVersion = "1.1.0"
+	integrationVersion = "1.2.0"
 
 	defaultHTTPTimeout = time.Second * 1
+
+	entityRemoteType = "apache"
```
Comment (ferranorriols) [inline src/apache.go]: check the entityType following the MMF documentation

File: src/apache.go [hunk 2]
```diff
@@ -29,21 +43,44 @@ func main() {
 	log.Debug("Starting Apache integration")
 	defer log.Debug("Apache integration exited")
 
-	integration, err := integration.New(integrationName, integrationVersion, integration.Args(&args))
+	var i *integration.Integration
```
Comment (ferranorriols) [inline src/apache.go]: Create a function which returns the integartion

File: src/apache.go [hunk 3]
```diff
@@ -52,7 +89,56 @@ func main() {
...
+}
+
+// parseStatusURL will extract the hostname and the port from the nginx status URL.
+func parseStatusURL(statusURL string) (hostname, port string, err error) {
+	u, err := url.Parse(statusURL)
+	if err != nil {
+		return
+	}
+
+	isHTTP := u.Scheme == httpProtocol || u.Scheme == httpsProtocol
```
Comment (ferranorriols) [inline src/apache.go]: avoid these boolean variables

File: src/metrics_test.go [hunk 4]
```diff
@@ -213,17 +213,17 @@ func TestGetBytes_InvalidDataKey(t *testing.T) {
 	}
 }
 
-var entity *integration.Entity
+var e *integration.Entity
```
Comment (ferranorriols) [inline src/metrics_test.go]: I would avoid having this global variable shared with different tests.

**Comment by ferranorriols** [comment]
I've seen in the original PR was using `integration2` package naming. Do you think we could change this and using the default `integration` package name?

File: src/metrics_test.go (hunk 4 above)
Comment (cristianciutea) [reply src/metrics_test.go]: I think this was added in order to avoid parsing the arguments multiple times. If I change this the sdk will panic

**Pull request merged**